compiler/src/vm/mod.rs

```diff
@@ -1,8 +1,854 @@
+                children, nursery, ..
+            }) => {
+                self.channels
+                    .remove(&nursery)
+                    .unwrap()
+                    .to_nursery()
+                    .unwrap();
+                for child_fiber in children.keys() {
+                    self.cancel(*child_fiber);
+                }
+            }
+            FiberTree::Try(Try { child, .. }) => self.cancel(child),
+        }
+    }
+
+    fn send_to_channel(
+        &mut self,
+        performing_fiber: Option<FiberId>,
+        channel_id: ChannelId,
+        packet: Packet,
+    ) {
+        let channel = match self.channels.get_mut(&channel_id) {
+            Some(channel) => channel,
+            None => {
+                // The channel was a nursery that died.
+                if let Some(fiber) = performing_fiber {
+                    let tree = self.fibers.get_mut(&fiber).unwrap();
+                    tree.as_single_mut().unwrap().fiber.panic(
+                        "the nursery is already dead because the parallel section ended"
+                            .to_string(),
+                    );
+                }
+                return;
+            }
+        };
+        match channel {
+            Channel::Internal(channel) => {
+                channel.send(&mut self.fibers, performing_fiber, packet);
+            }
+            Channel::External => self.push_external_operation(
+                channel_id,
+                Operation::Send {
+                    performing_fiber,
+                    packet,
+                },
+            ),
+            Channel::Nursery(parent_id) => {
+                info!("Nursery received packet {:?}", packet);
+                let parent_id = *parent_id;
+
+                match Self::parse_spawn_packet(packet) {
+                    Some((heap, closure_to_spawn, return_channel)) => {
+                        let child_id = self.fiber_id_generator.generate();
+                        self.fibers.insert(
+                            child_id,
+                            FiberTree::Single(Single {
+                                fiber: Fiber::new_for_running_closure(heap, closure_to_spawn, &[]),
+                                parent: Some(parent_id),
+                            }),
+                        );
+
+                        self.fibers
+                            .get_mut(&parent_id)
+                            .unwrap()
+                            .as_parallel_mut()
+                            .unwrap()
+                            .children
+                            .insert(child_id, ChildKind::SpawnedChild(return_channel));
+                    }
+                    None => self.finish_parallel(
+                        parent_id,
+                        Err("a nursery received an invalid message".to_string()),
+                    ),
+                }
+
+                InternalChannel::complete_send(&mut self.fibers, performing_fiber);
```

Comment:
If channels send something to a nursery, the send completes immediately after the new fiber is spawned on the nursery. This call in particular sets the status of the performing fiber back to `Running` so that it's considered as a candidate for further execution in the next `raw_run`.